express/controllers/cart.controller.ts

```diff
@@ -0,0 +1,83 @@
+import type { ICartEntity } from "../types/ICartEntity";
+import { Request, Response } from "express";
+import { CartRepository } from "../repositories/cart.repository";
+import Joi from "joi";
+import { IUserEntity } from "../types/IUserEntity";
+import { UserRepository } from "../repositories/user.repository";
+import { OrderRepository } from "../repositories/order.repository";
+
+export class CartController {
+
+    static findOrCreate(req: Request, res: Response) {
+        const userId = req.headers['x-user-id'];
+        const cart = CartRepository.findOrCreate(typeof userId === "string" ? userId : "");
+
+        res.statusCode = 200;
+        res.setHeader("Content-Type", "application/json");
+        res.send(JSON.stringify({ data: cart, error: null }));
+    }
+
+    static updateCart(req: Request, res: Response) {
+        const updateCartBodySchema = Joi.object({
+            productId: Joi.string(),
+            count: Joi.number()
+        })
+        const { error, value } = updateCartBodySchema.validate(req.body);
+
+        const userId = req.headers['x-user-id'];
+
+        if(error) {
+            res.statusCode = 400;
+            res.setHeader("Content-Type", "application/json");
+            res.send(JSON.stringify({ data: null, error: { message: "Products are not valid." } }));
+            return;
+        }
+
+        const cartSnapshot: ICartEntity | undefined = CartRepository.updateCart(value, typeof userId === "string" ? userId : "");
+        if(!cartSnapshot) {
+            res.statusCode = 404;
+            res.setHeader("Content-Type", "application/json");
+            res.send(JSON.stringify({ data: null, error: { message: "Cart was not found." } }));
```

Comment:
you constantly writing this 3 lines of code everywhere. why not make a function that abstract this part of code.
example: function response(status, message) {
res.statusCode = status;
            res.setHeader("Content-Type", "application/json");
            res.send(JSON.stringify({ data: null, error: { message } }));
}